product/interfaces/graphql/schema.graphql

```diff
@@ -145,6 +145,12 @@ type Commerce_ProductPriceInfo {
 
 }
 
+type Commerce_CategoryProductList {
+    category: Commerce_Category!
```

Comment:
Until now the "product" modul didnt know about the "category" modul. It is the other way around.
Cool would be a concept to search products by "any" filter like we have in the domain model..

Current dependency is:

Category -> Search
Category (->) Product -> Search

According to this the following idea:
Search Modul:
- Offer a "Commerce_SearchFilter interface" 
- Offers "Commerce_SearchMeta"
- Offers also implementations - e.g. a "KeyValueSearchFilter" ...

Product Modul:

Commerce:ProductSearchResult {
 products: [Commerce_Product!]
 meta: Commerce_SearchMeta
}

offer a Query:
- Commerce_ProductSearch(Commerce_SearchFilter[])


Category:
- Offers a Commerce_CategoryFilter (implements Commerce_SearchFilter)